tests/client/core/reducers/test_api.js

```diff
@@ -34,6 +38,26 @@ describe('api reducer', () => {
       { ...existingState, clientApp });
   });
 
+  it('stores the userAgent', () => {
+    const existingState = { bar: 'baz' };
+    const userAgent = userAgents.firefox[1];
+    const { browser, os } = UAParser(userAgent);
+
+    assert.deepEqual(
+      api(existingState, actions.setUserAgent(userAgent)),
+      { ...existingState, userAgent, userAgentInfo: { browser, os } });
+  });
+
+  it('allows empty userAgent', () => {
+    const existingState = { bar: 'baz' };
+    const userAgent = '';
```

Comment:
I think it would be good to add another test like this one that tries to pass an `undefined` `userAgent`